src/shapes/IText/ITextBehavior.ts

```diff
@@ -148,21 +148,26 @@ export abstract class ITextBehavior<
     });
   }
 
+  /**
+   * changes the cursor from visible to invisible
+   */
   private _tick(delay?: number) {
     this._currentTickState = this._animateCursor({
-      toValue: 1,
-      duration: this.cursorDuration,
-      delay,
+      toValue: 0,
+      duration: this.cursorDuration / 2,
+      delay: delay || 100,
       onComplete: this._onTickComplete,
```

Comment:
I feel strongly against recursive/looping methods calling each other.
IMO a recursive/looping method should call ONLY itself.
Spreading recursion across multiple methods is confusing and is prone for inifinte loops.

Reply:
Therefore, I suggest to make it a single function accepting a boolean flagging the state

Reply:
Just open an issue with label enanchement, the first one that can do it will do it if there is a real improvement of some kind, i can't see one that make me jump at it.
If you look it from a different point of view tick and tickComplete are 2 wrappers around animateCursor, and is just animateCursor calling itself with configuration A or B.

The issue with not having 2 methods is that you have to use that boolean to determine the 3 conditions every time you call the single method and you don't have a single place where to determine the first delay ( delay || 100 ) so it will be harder to track which is the start of the animation.
Today that is done implicitly by calling _onTick bound without parameters by onComplete

Reply:
I have other questions on this code more than the double methods:
- why do we need to track the 2 animations with 2 separate states ( they are running only on sequence )
- why onTickComplete does a pre abort check while onTick doesnt

but also those are out of scope for the animation timing tweak

Reply:
here is a tracking issue https://github.com/fabricjs/fabric.js/issues/9835